### ISSUE TITLE
Mouse out-of-bounds assertion when switching to fullscreen

### ISSUE THREAD

**Issue by kcgen** [dosbox-staging/dosbox-staging#2107]
@FeralChild64 , this looks like an interesting corner case, but could be an SDL bug too. I'm using SDL 2.24.0. It's really nice to see the assert catching this!

```
src/hardware/mouse/mouse.cpp:86:
  Assertion 'clipping * 2 < resolution' failed.
```

Reproduced using a debug-build of the `main` branch launching `dosbox --noprimaryconf` plus this local `dosbox.conf`:

```ini
[sdl]
fullscreen = false
output = surface
```

Then hit `Alt+Enter` to switch to fullscreen, which should trigger the assert.

**Comment by GranMinigun** [comment]
I can sometimes trigger this assert by resizing window with either texture or OpenGL outputs. More often when the window is smaller than emulated video mode, I think.

**Comment by FeralChild64** [comment]
@GranMinigun Indeed, when the window is resized to be much smaller than guest display, with odd proportions, the values are strange, too. But the core reason is a separate issue - most likely a long-standing one.
